src/kagglehub/models_helpers.py

```diff
@@ -0,0 +1,227 @@
+MODEL_METADATA_FILE = "model-metadata.json"
+MODEL_INSTANCE_METADATA_FILE = "model-instance-metadata.json"
+
+
+def parse(string):
+    time_formats = ["%Y-%m-%dT%H:%M:%S", "%Y-%m-%dT%H:%M:%SZ", "%Y-%m-%dT%H:%M:%S.%f", "%Y-%m-%dT%H:%M:%S.%fZ"]
+    for t in time_formats:
+        try:
+            result = datetime.strptime(string[:26], t).replace(microsecond=0)
+            return result
+        except:  # noqa: E722, S110
+            pass
+    return string
+
+
+class DirectoryArchive:
+    def __init__(self, fullpath, format):  # noqa: A002
+        self._fullpath = fullpath
+        self._format = format
+        self.name = None
+        self.path = None
+
+    def __enter__(self):
+        self._temp_dir = tempfile.mkdtemp()
+        _, dir_name = os.path.split(self._fullpath)
+        self.path = shutil.make_archive(os.path.join(self._temp_dir, dir_name), self._format, self._fullpath)
+        _, self.name = os.path.split(self.path)
+        return self
+
+    def __exit__(self, *args):
+        shutil.rmtree(self._temp_dir)
+
+
+class File(object):  # noqa: UP004
+    def __init__(self, init_dict):
+        parsed_dict = {k: parse(v) for k, v in init_dict.items()}
+        self.__dict__.update(parsed_dict)
+        self.size = File.get_size(self.totalBytes)
+
+    def __repr__(self):
+        return self.ref
+
+    @staticmethod
+    def get_size(size, precision=0):
+        suffixes = ["B", "KB", "MB", "GB", "TB"]
+        suffix_index = 0
+        while size >= 1024 and suffix_index < 4:  # noqa: PLR2004
+            suffix_index += 1
+            size /= 1024.0
+        return "%.*f%s" % (precision, size, suffixes[suffix_index])
+
+
+def _upload_blob(file_path):
+    data = {
+        "type": "model",
+        "name": os.path.basename(file_path),
+        "contentLength": os.path.getsize(file_path),
+        "lastModifiedEpochSeconds": int(os.path.getmtime(file_path)),
+    }
+    api_client = KaggleApiV1Client()
+    response = api_client.post("/blobs/upload", data=data)
+    if "error" in response and response["error"] != "":
+        raise BackendError(response["error"])
+
+    with open(file_path, "rb") as f:
+        file_data = f.read()
+
+    headers = {"Content-Type": "application/octet-stream"}
+    try:
+        requests.put(response["createUrl"], data=file_data, headers=headers, timeout=10)
+    except requests.exceptions.Timeout:
+        # Handle the timeout exception
+        logger.error("Request timed out")
+
+    if "error" in response and response["error"] != "":
+        raise BackendError(response["error"])
+
+    return response["token"]
+
+
+def upload_files(folder, quiet=False, dir_mode="skip"):  # noqa: FBT002
+    """upload files in a folder
+        Parameters
+    ==========
+    request: the prepared request
+    folder: the folder to upload from
+    quiet: suppress verbose output (default is False)
+    """
+    tokens = []
+    for file_name in os.listdir(folder):
```

Comment:
You should throw an exception if the user is trying to upload more than 50 files. 
Add a TODO to handle this case by first zipping the content of the folder (and address in separate PR).

Make sure to walk down the tree (count files in each subdirectories). See my comment below.